cadasta/gui/tools/wizard/step_project_creation01.py

```diff
@@ -0,0 +1,210 @@
+
+        :returns: The step to be switched to
+        :rtype: WizardStep instance or None
+        """
+        new_step = self.parent.step_project_creation02
+        return new_step
+
+    def get_available_organisations(self):
+        """Get available organisations and load it to
+           organisation combo box.
+        """
+        LOGGER.info('Getting organisations')
+        self.get_organisation_button.setEnabled(False)
+        status, results = self.organisation.all_organizations()
+        self.get_organisation_button.setEnabled(True)
+        if status:
+            self.organisation_box.clear()
+            self.parent.organisations_list = results
+            for organisation in results:
+                self.organisation_box.addItem(
+                    organisation['name'],
+                    organisation
+                )
+        else:
+            self.message_bar.pushWarning(
+                    self.tr('Error'),
+                    self.tr(results)
+            )
+
+    def get_available_layers(self):
+        """Get layer from qgis and load it to combo box."""
+        layers = self.parent.iface.legendInterface().layers()
+        for layer in layers:
+            self.qgis_layer_box.addItem(
+                    layer.name(),
+                    layer
+            )
+
+    def all_data(self):
+        """Return all data from this step.
+
+        :returns: step 1 data
+        :rtype: dict
+        """
+        data = dict()
+        data['organisation'] = self.selected_organisation()
+        data['project_name'] = self.project_name()
+        data['project_url'] = self.project_url()
+        data['contact'] = {
+            'name': self.contact_name_text.displayText(),
+            'phone': self.contact_phone_text.displayText(),
+            'email': self.contact_email_text.displayText()
+        }
+        data['description'] = self.project_description_text.toPlainText()
+        data['private'] = self.is_project_private.isChecked()
+
+        # Get extent
+        layer = self.selected_layer()
+        if self.use_layer_extents.isChecked():
+            # Layer extent
+            extent = layer.extent()
+        else:
+            # Canvas extent
+            extent = self.parent.iface.mapCanvas().extent()
+
+        geometry = QgsGeometry().fromRect(extent)
+        data['extent'] = geometry.exportToGeoJSON()
+
+        # Save layer to geojson format
+        output_file = '/tmp/project.json'
+
+        result = QgsVectorFileWriter.writeAsVectorFormat(
+            layer,
+            output_file,
+            'utf-8',
+            layer.crs(),
+            'GeoJson'
+        )
+
+        if result == QgsVectorFileWriter.NoError:
+            LOGGER.debug('Wrote layer to geojson: %s' % output_file)
+            with open('/tmp/project.json') as json_data:
+                layer_data = json.load(json_data)
+                data['layer'] = layer_data
+            os.remove(output_file)
```

Comment:
Yes, is there any better approach for this? I already looking in google and asking the guys, and this is the best I can do.